.github/workflows/test.yml

```diff
@@ -51,12 +51,14 @@ jobs:
     runs-on: ubuntu-22.04
     timeout-minutes: 20
     steps:
-      - uses: actions/setup-go@v3
-        with:
-          go-version: 1.19.x
       - uses: actions/checkout@v3.3.0
         with:
           fetch-depth: 1
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.19.x
+          check-latest: true
+          cache: true
```

Comment:
nit: Why not enable this in other tests? (e.g. `lint`, `project`)?

Reply:
I did not realize that both the checks did go build; i.e the main reason.

golangci is already cached: https://github.com/containerd/nerdctl/actions/caches
I think having in one place is sufficient: See: https://github.com/containerd/nerdctl/actions/runs/4097163654/jobs/7065337105
Likely because a workflow is tied to a single runner.
https://github.com/containerd/nerdctl/actions/runs/4097163654/jobs/7065337009

I can make the change if you prefer.